packages/framework/presence/src/presence.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ConnectedClientId } from "./baseTypes.js";
+import type { NotificationsManager } from "./notificationsManager.js";
+import type {
+	PresenceStates,
+	PresenceWorkspaceAddress,
+	PresenceStatesSchema,
+} from "./types.js";
+
+import type { ISubscribable } from "@fluid-experimental/presence/internal/events";
+import type { InternalTypes } from "@fluid-experimental/presence/internal/exposedInternalTypes";
+
+/**
+ * A client within a Fluid session (period of container connectivity to service).
+ *
+ * @sealed
+ * @alpha
+ */
+export interface ISessionClient {
+	/**
+	 * Get current client connection id.
+	 *
+	 * @returns Current client connection id.
+	 *
+	 * @remarks
+	 * Connection id will change on reconnection.
+	 */
+	currentClientId(): ConnectedClientId;
+}
+
+/**
+ * @sealed
+ * @alpha
+ */
+export interface PresenceEvents {
+	/**
+	 * Raised when new client joins session.
+	 *
+	 * @eventProperty
+	 */
+	attendeeJoined: (attendee: ISessionClient) => void;
+
+	/**
+	 * Raised when client appears disconnected from session.
+	 *
+	 * @eventProperty
+	 */
+	attendeeDisconnected: (attendee: ISessionClient) => void;
+
+	/**
+	 * Raised when a workspace is activated within the session.
+	 */
+	workspaceActivated: (
```

Comment:
How visible is the "workspace" concept intended to be to customers?  From your spec it looked like it would be an internal concept only?

I'm also not exactly sure what "activated" means here.  Do they ever "deactivate"?

Reply:
Workspaces are the main point of interaction for the new feature. A workspace evolved as the term to cover both States and Notifications that were broken apart. Implementation-wise States is superset of functionality and supports notifications. Notifications is an instance of States that only allows Notification entries.

No deactivation. All new since the design after thinking about things.
Workspace activation means first time this client sees activity for a workspace. This will enable a client to be passive about listening to states/notifications until someone decides to send something.